src/main/scala/au/com/agiledigital/toolform/model/Project.scala

```diff
@@ -0,0 +1,114 @@
+package au.com.agiledigital.toolform.model
+
+import au.com.agiledigital.toolform.model.ReferenceType.ReferenceType
+import pureconfig.ConfigConvert
+
+/**
+  * Configuration for a project.
+  *
+  * @param id         the id of the project.
+  * @param name       the name of the project.
+  * @param components the components that will be built and combined to form the projects.
+  * @param resources  the resources that must be supplied to the project to run.
+  * @param topology   the network topology of the environment explains how elements will be related/connected.
+  * @param volumes    the volumes that will be mounted into components.
+  */
+final case class Project(id: String,
+                         name: String,
+                         components: Map[String, Component],
+                         resources: Map[String, Resource],
+                         topology: Topology,
+                         volumes: Option[Seq[Volume]],
```

Comment:
Lets make this just a `Seq[Volume]` (and switch to the immutable Seq) on the next PR.